src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CertificatePolicy.cs

```diff
@@ -0,0 +1,378 @@
+            return reader.ReadInteger();
+        }
+
+        private static void ReadCertPolicyConstraingsExtension(X509Extension extension, CertificatePolicy policy)
+        {
+            DerSequenceReader reader = new DerSequenceReader(extension.RawData);
+
+            while (reader.HasData)
+            {
+                switch (reader.PeekTag())
+                {
+                    case 0x80:
+                        policy.RequireExplicitPolicyDepth = reader.ReadInteger();
+                        break;
+                    case 0x81:
+                        policy.InhibitMappingDepth = reader.ReadInteger();
+                        break;
+                    default:
+                        if (extension.Critical)
+                        {
+                            // If an unknown value is read, but we're marked as critical,
+                            // then we don't know what we're doing and MUST fail validation
+                            // (RFC 3280).
+                            // If it isn't critical then it means we're allowed to be ignorant
+                            // of data defined more recently than we understand.
+                            throw new CryptographicException();
+                        }
+
+                        break;
+                }
+            }
+        }
+
+        private static ISet<string> ReadExtendedKeyUsageExtension(X509Extension extension)
+        {
+            X509EnhancedKeyUsageExtension ekusExtension = (X509EnhancedKeyUsageExtension)extension;
+            HashSet<string> oids = new HashSet<string>();
+
+            foreach (Oid oid in ekusExtension.EnhancedKeyUsages)
+            {
+                oids.Add(oid.Value);
+            }
+
+            return oids;
+        }
+
+        private static ISet<string> ReadCertPolicyExtension(X509Extension extension)
+        {
+            DerSequenceReader reader = new DerSequenceReader(extension.RawData);
+            HashSet<string> policies = new HashSet<string>();
+
+            while (reader.HasData)
+            {
+                DerSequenceReader policyInformation = reader.ReadSequence();
+                policies.Add(policyInformation.ReadOidAsString());
+
+                // There is an optional policy qualifier here, but it is for information
+                // purposes, there is no logic that would be changed.
+
+                // Since reader (the outer one) has already skipped past the rest of the
+                // sequence we don't particularly need to drain out here.
+            }
+
+            return policies;
+        }
+
+        private static LowLevelList<CertificatePolicyMapping> ReadCertPolicyMappingsExtension(X509Extension extension)
+        {
+            DerSequenceReader reader = new DerSequenceReader(extension.RawData);
+            LowLevelList<CertificatePolicyMapping> mappings = new LowLevelList<CertificatePolicyMapping>();
```

Comment:
Why ```LowLevelList<>``` instead of ```List<>```?  I'd thought we just used ```LowLevelSomeType``` when we couldn't reference the appropriate library containing ```SomeType``` and so instead included a copy of the source for it.